### PR TITLE
allow the users seeing their submitted application

### PR DESCRIPTION
**Pull request by ArabellaJi**

### CODE REVIEW

File: src/services/housing.ts
Comment (andrew-wzj) [inline src/services/housing.ts:1]: I see you added a few constructors for accessing users' preference, preferred hall, and roommates. But I don't see related code here for allowing the users seeing their submitted application

File: src/views/HousingLottery/studentView/PreferredHall/HallSlotComponent/index.jsx
Comment (andrew-wzj) [inline src/views/HousingLottery/studentView/PreferredHall/HallSlotComponent/index.jsx:1]: If you trying to make the selected preferred hall dynamically changed, there is the another merged PR https://github.com/gordon-cs/gordon-360-ui/pull/2202 (that I reviewed and tested works fine)

File: src/views/HousingLottery/studentView/PreferredHall/index.jsx
Comment (andrew-wzj) [inline src/views/HousingLottery/studentView/PreferredHall/index.jsx:1]: see comment above

**Review by andrew-wzj — COMMENTED** [review]
The commit message wasn't quite matching with the code change

**Review by andrew-wzj — COMMENTED** [review]
Update explanation for this PR: the code change is for making the application page to store the submitted application for applicant for allowing him or her to access the same application again if he or she wants to edit the saved application, and once the submit button is pressed again, the corresponding application data would be overwritten in the database

File: src/views/HousingLottery/studentView/PreferredHall/HallSlotComponent/index.jsx
Comment (ArabellaJi) [reply src/views/HousingLottery/studentView/PreferredHall/HallSlotComponent/index.jsx:1]: #2202 The link you provided above seems like the link over here lol. Do you mean to provide another link?

File: src/services/housing.ts
Comment (ArabellaJi) [reply src/services/housing.ts:1]: The updated code in the two index.jsx files are the codes allowing the users seeing their submitted application.

**Review by andrew-wzj — APPROVED** [review]

**Pull request merged**